fiftyone/utils/ultralytics.py

```
@@ -270,3 +289,105 @@ def _to_detections(self, df):
+    """YOLO detection model wrapper for FiftyOne.
+
+    Args:
+        config: a dict which has the `ultralytics.YOLO` Detection model as the
+            value for the key `model`
+    """
+
+    def predict(self, arg):
+        image = Image.fromarray(arg)
+        predictions = self.model(image, verbose=False)
+        return to_detections(predictions[0])
+
+
+class FiftyOneYOLOSegmentationModel(FiftyOneYOLOModel):
+    """YOLO segmentation model wrapper for FiftyOne.
+
+    Args:
+        config: a dict which has the `ultralytics.YOLO` Segmentation model as
+            the value for the key `model`
+    """
+
+    def predict(self, arg):
+        image = Image.fromarray(arg)
+        predictions = self.model(image, verbose=False)
+        return to_instances(predictions[0])
+
+
+class FiftyOneYOLOPoseModel(FiftyOneYOLOModel):
+    """YOLO pose model wrapper for FiftyOne.
+
+    Args:
+        config: a dict which has the `ultralytics.YOLO` Pose model as the value
+            for the key `model`
+    """
+
+    def predict(self, arg):
+        image = Image.fromarray(arg)
+        predictions = self.model(image, verbose=False)
+        return to_keypoints(predictions[0])
+
+
+def _convert_yolo_detection_model(model):
+    config = {"model": model}
+    return FiftyOneYOLODetectionModel(config)
+
+
+def _convert_yolo_segmentation_model(model):
+    config = {"model": model}
+    return FiftyOneYOLOSegmentationModel(config)
+
+
+def _convert_yolo_pose_model(model):
+    config = {"model": model}
+    return FiftyOneYOLOPoseModel(config)
+
+
+def _convert_yolo_model(model):
+    if isinstance(model.model, ultralytics.nn.tasks.SegmentationModel):
+        return _convert_yolo_segmentation_model(model)
+    elif isinstance(model.model, ultralytics.nn.tasks.PoseModel):
+        return _convert_yolo_pose_model(model)
+    elif isinstance(model.model, ultralytics.nn.tasks.DetectionModel):
+        return _convert_yolo_detection_model(model)
+
+    return model
```

Better to raise an error here saying that we cannot convert this model to an FO `Model`?

---

Totally agree!